runtime/src/main/java/org/corfudb/runtime/clients/LayoutHandler.java

```diff
@@ -52,46 +59,116 @@ public LayoutClient getClient(long epoch, UUID clusterID) {
             .generateErrorHandlers(MethodHandles.lookup(), this);
 
+    /**
+     * TODO: Remove this method when Management RPC get landed.
+     */
+    @Deprecated
+    @ClientHandler(type = CorfuMsgType.LAYOUT_NOBOOTSTRAP)
+    private static Object handleNoBootstrap(CorfuMsg msg,
+                                            ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
+        throw new NoBootstrapException();
+    }
 
+    /**
+     * TODO: Remove this method when Management RPC get landed.
+     */
+    @Deprecated
     @ClientHandler(type = CorfuMsgType.LAYOUT_RESPONSE)
     private static Object handleLayoutResponse(CorfuMsg msg,
                                                ChannelHandlerContext ctx, IClientRouter r) {
         return ((LayoutMsg) msg).getLayout();
     }
 
-    @ClientHandler(type = CorfuMsgType.LAYOUT_PREPARE_ACK)
-    private static Object handleLayoutPrepareAck(CorfuPayloadMsg<LayoutPrepareRequest> msg,
-                                                 ChannelHandlerContext ctx, IClientRouter r) {
-        return msg.getPayload();
+    /**
+     * Handle a layout response from the server.
+     *
+     * @param msg The layout response message.
+     * @param ctx The context the message was sent under.
+     * @param r A reference to the router.
+     * @return {@link Layout} sent back from server.
+     */
+    @ResponseHandler(type = PayloadCase.LAYOUT_RESPONSE)
+    private static Object handleLayoutResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                               IClientRouter r) {
+        LayoutResponseMsg layoutResponse = msg.getPayload().getLayoutResponse();
+        RpcCommon.LayoutMsg layoutMsg = layoutResponse.getLayout();
+
+        return CorfuProtocolCommon.getLayout(layoutMsg);
     }
 
-    @ClientHandler(type = CorfuMsgType.LAYOUT_NOBOOTSTRAP)
-    private static Object handleNoBootstrap(CorfuMsg msg,
-                                            ChannelHandlerContext ctx, IClientRouter r)
-            throws Exception {
-        throw new NoBootstrapException();
+    /**
+     * Handle a prepare layout response from the server.
+     *
+     * @param msg The prepare layout response message.
+     * @param ctx The context the message was sent under.
+     * @param r A reference to the router.
+     * @return {@link LayoutPrepareResponse} if ACK, throw an {@link OutrankedException} if REJECT.
+     */
+    @ResponseHandler(type = PayloadCase.PREPARE_LAYOUT_RESPONSE)
+    private static Object handlePrepareLayoutResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                                      IClientRouter r) {
+        PrepareLayoutResponseMsg prepareLayoutMsg = msg.getPayload().getPrepareLayoutResponse();
+        boolean prepared = prepareLayoutMsg.getPrepared();
+        long rank = prepareLayoutMsg.getRank();
+        Layout layout = CorfuProtocolCommon.getLayout(prepareLayoutMsg.getLayout());
```

Comment:
static methods make code less maintainable/testable

Reply:
Sure, but in this case its a It's a stateless helper method. Anyways, if you still want this to change, the please propose something more actionable.